### ISSUE TITLE
Add new EFI features to the typescript definition file

### ISSUE THREAD

**Issue by leahanderson1** [eaglerforge/example-typescript-mod#4]
a bunch of new EFI features need to be added to the definition file so it can properly use intellisense

**Comment by leahanderson1** [comment]
oh we also need to update the API docs iirc

**Comment by leahanderson1** [comment]
nvm the docs are in the EFI repo but it'd still be helpful to link them there

**Comment by leahanderson1** [comment]
or we could just move those to eaglerforge.github.io

**Comment by radmanplays** [comment]
> or we could just move those to eaglerforge.github.io

I agree

**Comment by leahanderson1** [comment]
yeah i think it'd be better to have the docs just at eaglerforge.github.io

**Comment by leahanderson1** [comment]
we should also add descriptions to the d.ts documentation

**Comment by ZXMushroom63** [comment]
> a bunch of new EFI features need to be added to the definition file so it can properly use intellisense

about that...........
EFI's structure is a little bit different from legacy EaglerForge, as instead of having code to manually expose each property, its all based on directly interfacing with eaglercraft at runtime. we'd pretty much need to convert all of minecraft into a .d.ts file (which I've got no clue how to use)

After I'm done doing custom meshes and whatnot I can try writing a script to traverse the modapi object and generate a .d.ts from it, but it's still not gonna be fun.

**Comment by STPv22** [comment]
Is there any way I can still make ts mods with the new API?

**Comment by ZXMushroom63** [comment]
> Is there any way I can still make ts mods with the new API?

yes, you just wont get autocomplete features

**Comment by leahanderson1** [comment]
typescript mods are better for organization as you can have stuff in different files :3 thats why I would recommend using it over just writing a .js file

**Comment by STPv22** [comment]
`declare var ModAPI: any`, right?

**Comment by STPv22** [comment]
> typescript mods are better for organization as you can have stuff in different files :3 thats why I would recommend using it over just writing a .js file

ye that's why  want to try it :3

**Comment by leahanderson1** [comment]
> `declare var ModAPI: any`, right?

pretty sure that would work too

**Comment by leahanderson1** [comment]
try it out

**Comment by STPv22** [comment]
any update on how finished the definitions file is?

**Comment by ZXMushroom63** [comment]
didn't block2222 pretty much finish this

**Comment by STPv22** [comment]
I mean, it looks finished, I just wanna double check
